packages/compartment-mapper/test/optional.test.js

```diff
@@ -15,7 +15,8 @@ const fixtureOptionalDepsCjs = new URL(
 
 scaffold(
   'optionalDependencies/esm',
-  // this test fails because it relies on dynamic import
+  // fails for archives because dynamic import cannot reach modules not
+  // discovered during archival
```

Comment:
does dynamic require have the same limitation?

Reply:
Yes.